src/lib/vocab/policy/policyHelpers.server.ts

```diff
@@ -75,12 +74,18 @@ export const checkEntityOwnership = async (
 		return {ok: false, status: 400, message: 'unable to process non-existing entities'};
 	}
 
-	for (const entity of entities) {
-		if (entity.persona_id !== actor_id) {
-			//TODO probably want some caching here if we're gonna have particularly large batches of updates/dels
+	const spaceIds = new Set(
+		entities.map((e) => (e.persona_id === actor_id ? null : e.space_id)).filter(Boolean),
```

Comment:
problem here is TypeScript doesn't understand the `filter(Boolean)`, so adding the `!` is a hack, but it means you can delete the `if (space_id) {` guard below

```suggestion
		entities.map((e) => (e.persona_id === actor_id ? null! : e.space_id)).filter(Boolean),
```